daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/ClosureConversionStackSafe.scala

```diff
@@ -0,0 +1,368 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.speedy
+
+/** Closure Conversion (Phase of the speedy compiler pipeline)
+  *
+  * This compilation phase transforms from SExpr0 to SExpr1.
+  */
+
+import com.daml.lf.data.Ref
+
+import com.daml.lf.speedy.SExpr.SCasePat
+import com.daml.lf.speedy.{SExpr0 => source}
+import com.daml.lf.speedy.{SExpr1 => target}
+
+import scala.annotation.tailrec
+
+private[speedy] object ClosureConversionStackSafe {
+
+  private[speedy] def closureConvert(source0: source.SExpr): target.SExpr = {
+
+    // This is the 'Env' implementation from the pre-stack-safe closure-conversion code.
```

Comment:
Referring to code that we deleted isn’t all that useful. Let’s focus the comment on how the issues and how we want to change it rather than where it originates from.

Reply:
ok